### PR TITLE
OBJ-156 Kafka beans configuration

### PR DESCRIPTION
**Pull request by mattch1**
Beans config, rest client and test for rest client.

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/api/strikeoffobjections/email/KafkaRestClient.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+    public KafkaRestClient(RestTemplate restTemplate) {
+        this.restTemplate = restTemplate;
+    }
+
+    public byte[] getSchema(String schemaRegistryUrl, String emailSchemaUri) {
+        String schemaUrl = String.format("%s%s", schemaRegistryUrl, emailSchemaUri);
+        HttpHeaders headers = new HttpHeaders();
+        HttpEntity<String> entity = new HttpEntity<>(headers);
+        ResponseEntity<byte[]> response = restTemplate.exchange(schemaUrl, HttpMethod.GET, entity, byte[].class);
```
Comment (markpit) [inline src/main/java/uk/gov/companieshouse/api/strikeoffobjections/email/KafkaRestClient.java:21]: does it work if set the exchange type to String.class? you might be able to get rid of 
String schemaJson = new JSONObject(new String(bytes)).getString("schema");
in KafkaConfiguration then.

Comment (markpit) [reply src/main/java/uk/gov/companieshouse/api/strikeoffobjections/email/KafkaRestClient.java:21]: i mean the new String(bytes) part

File: src/test/java/email/KafkaRestClientUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    public void setUp() {
+        restTemplate = Mockito.mock(RestTemplate.class);
+        restClient = new KafkaRestClient(restTemplate);
+        schemaUrl = String.format("%s%s", schemaRegistryUrl, emailSchemaUri);
+        response = new ResponseEntity<>(HttpStatus.OK);
+        when(restTemplate.exchange(eq(schemaUrl), eq(HttpMethod.GET), any(), eq(byte[].class))).thenReturn(response);
+    }
+
+    @Test
+    public void testExchangeHasBeenCalled() {
```
Comment (markpit) [inline src/test/java/email/KafkaRestClientUnitTest.java:38]: check the method returns the correct value as well.

File: src/main/java/uk/gov/companieshouse/api/strikeoffobjections/email/KafkaRestClient.java (hunk 1 above)
Comment (mattch1) [reply src/main/java/uk/gov/companieshouse/api/strikeoffobjections/email/KafkaRestClient.java:21]: Really good question -I don't know - hard to test until all subtaks are merged together.

Comment (markpit) [reply src/main/java/uk/gov/companieshouse/api/strikeoffobjections/email/KafkaRestClient.java:21]: we can try at a later date, might be a good little improvement if it works

File: src/test/java/email/KafkaRestClientUnitTest.java (hunk 2 above)
Comment (mattch1) [reply src/test/java/email/KafkaRestClientUnitTest.java:38]: Not sure what can be meaningfully done here with this - any suggestions would be welcome.

File: src/main/java/uk/gov/companieshouse/api/strikeoffobjections/config/ApplicationConfig.java [hunk 3]
```diff
@@ -3,7 +3,9 @@
 import org.springframework.boot.web.client.RestTemplateBuilder;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
+import org.springframework.http.client.HttpComponentsClientHttpRequestFactory;
```
Comment (ICoxCH) [inline src/main/java/uk/gov/companieshouse/api/strikeoffobjections/config/ApplicationConfig.java:6]: Please check the sonar report before putting code up for review, this import is unused

File: src/test/java/email/KafkaRestClientUnitTest.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        restClient.getSchema(schemaRegistryUrl, emailSchemaUri);
```
Comment (markpit) [inline src/test/java/email/KafkaRestClientUnitTest.java:39]:
```suggestion
        byte[] schema = restClient.getSchema(schemaRegistryUrl, emailSchemaUri);
        assertEquals(response.body(), schema);
```
looks like you'd have to set a dummy body on the response as well.

File: src/main/java/uk/gov/companieshouse/api/strikeoffobjections/config/ApplicationConfig.java (hunk 3 above)
Comment (mattch1) [reply src/main/java/uk/gov/companieshouse/api/strikeoffobjections/config/ApplicationConfig.java:6]: Seen that - taken it out.

File: src/test/java/email/KafkaRestClientUnitTest.java (hunk 4 above)
Comment (mattch1) [reply src/test/java/email/KafkaRestClientUnitTest.java]: Added this an run it - passes.

File: src/test/java/email/KafkaRestClientUnitTest.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+    private KafkaRestClient restClient;
+    private RestTemplate restTemplate;
+
+    @BeforeEach
+    public void setUp() {
+        restTemplate = Mockito.mock(RestTemplate.class);
+        restClient = new KafkaRestClient(restTemplate);
+        schemaUrl = String.format("%s%s", schemaRegistryUrl, emailSchemaUri);
+        response = new ResponseEntity<>(HttpStatus.OK);
```
Comment (ICoxCH) [inline src/test/java/email/KafkaRestClientUnitTest.java:34]: Think Mark mentioned it, but as we are asserting on the response.body we should probably give it some content

Comment (markpit) [reply src/test/java/email/KafkaRestClientUnitTest.java:34]: yes, in the constructor args, you should be able to add the body as the first param

Comment (mattch1) [reply src/test/java/email/KafkaRestClientUnitTest.java]: Added your response body

**Pull request merged**